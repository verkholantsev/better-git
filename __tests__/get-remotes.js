// @flow

import fs from 'fs';
import path from 'path';
import promisify from 'util.promisify';

import getRemotes from '../src/get-remotes';

const readFile = promisify(fs.readFile);

describe('git.getRemotes()', () => {
    it('should work correctly', async () => {
        const fixturePath = path.resolve(__dirname, '__fixtures__', 'git-remote-v');
        const git = async () => await readFile(fixturePath, { encoding: 'utf8' });

        expect(await getRemotes(git)).toMatchSnapshot();
    });
});
