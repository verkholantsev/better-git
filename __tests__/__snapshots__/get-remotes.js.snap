// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`git.getRemotes() should work correctly 1`] = `
Array [
  Object {
    "name": "origin",
    "refs": Object {
      "fetch": "git@github.com:verkholantsev/yarn.git",
      "push": "git@github.com:verkholantsev/yarn.git",
    },
  },
  Object {
    "name": "upstream",
    "refs": Object {
      "fetch": "git@github.com:yarnpkg/yarn.git",
      "push": "git@github.com:yarnpkg/yarn.git",
    },
  },
]
`;
